app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    def wrapper(*args: tuple) -> Callable:
+        if args in cache_store:
```

Comment:
Using mutable types like lists or dictionaries as keys in a dictionary will raise a TypeError. Ensure that the arguments passed to the function are hashable, or consider converting them to a hashable type before using them as keys.